File: josephus.py
import sys

def eliminate(nPeople, kill, nSurvivor = 1) :
	left = 0
	while len(nPeople) > nSurvivor :
		num = kill - left

		if(num > len(nPeople)) :
			while num > len(nPeople) :
				num = num - len(nPeople)

		while num <= len(nPeople) :
			nPeople[num - 1] = 0
			num = num + kill
		
		try :
			left = len(nPeople[(((len(nPeople) - 1) - nPeople[::-1].index(0)) + 1):])
		except ValueError :
			pass

		nPeople = list(filter((0).__ne__, nPeople))
	return nPeople

people = input('Enter the initial number of people : ')
killfreq = input('Enter the person number to be killed : ')
survivor = input('Enter number of survivors : ')

if int(survivor) == 0 :
	print("\n\nEveryone is dead!")
	sys.exit()
elif int(survivor) < 0 :
	survivor = 1

peopleList = [int(i+1) for i in range (int(people))]

survivorList = eliminate(peopleList, int(killfreq), int(survivor))

if len(survivorList) == 1 :
	print("\n\nThe survivor is number " + str(survivorList[0]))
else :
	print("\n\nThe survivor numbers are : ")
	for i in survivorList :
		print(i, end = " ")

File: josephus.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    unsigned int people, killfreq, survivor;
    unsigned int left = 0, num;

    std::cout << "Enter the initial number of people : ";
    std::cin >> people;
    std::cout << "Enter the person number to be killed : ";
    std::cin >> killfreq;
    std::cout << "Enter the number of survivor : ";
    std::cin >> survivor;

    if(survivor == 0) {
        std::cout << "\nEveryone is dead";
        return 0;
    } else if(survivor < 0) {
        survivor = 1;
    }

    std::vector<int> peopleList;
    peopleList.reserve(people);

    for(unsigned int i = 1; i <= people; ++i) {
        peopleList.push_back(i);
    }

    while(peopleList.size() > survivor) {
            num = killfreq - left;

            if(num > peopleList.size()) {
                while(num > peopleList.size()) {
                    num = num - peopleList.size();
                }
            }

            while(num <= peopleList.size()) {
                peopleList[num - 1] = 0;
                num = num + killfreq;
            }

            for(int j = peopleList.size() - 1; j >= 0; j--) {
                    if (peopleList[j] == 0) {
                        left = peopleList.size() - (j + 1);
                        break;
                    }
            }

            peopleList.erase(std::remove(peopleList.begin(), peopleList.end(), 0), peopleList.end());
    }

    if(survivor == 1) {
        std::cout << "\nThe survivor is number " << peopleList[0];
    } else {
        std::cout << "\nThe survivors are numbers ";
        for(unsigned int k = 0; k < peopleList.size(); ++k) {
            std::cout << peopleList[k] << " ";
        }
    }

    return 0;
}
